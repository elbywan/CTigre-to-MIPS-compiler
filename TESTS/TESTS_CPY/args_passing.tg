/* This program show that for:
     - simple variable => we have by-value calling
     - structured vars => we have by-reference calling
 */
type intArray = array of int in
var x := 42 in
var tab := intArray[3] of -1 in

function f(x, y) =
  x := x+10;
  y[0] := x;
  print("inside x:");printint(x);print("\n");
  print("inside tab[0]:");printint(y[0]);print("\n")
in

print("before x:");printint(x);print("\n");
print("before tab[0]:");printint(tab[0]);print("\n");
f(x,tab);
print("after x:");printint(x);print("\n");
print("after tab[0]:");printint(tab[0]);print("\n")