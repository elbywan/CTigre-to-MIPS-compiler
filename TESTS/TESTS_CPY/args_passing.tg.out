before x:42
before tab[0]:-1
inside x:52
inside tab[0]:52
after x:42
after tab[0]:52

